pkg/apis/crd/v1alpha1/types.go

```diff
@@ -393,6 +393,12 @@ type Rule struct {
 	// empty, this rule matches all protocols supported.
 	// +optional
 	Protocols []NetworkPolicyProtocol `json:"protocols,omitempty"`
+	// Set of Layer 7 protocols matched by the rule. If this field is set, action can only be Allow.
+	// When this field is used in a rule, any traffic matching the other criteria of the rule (typically the 5-tuple)
```

Comment:
other criteria -> L4 criteria?

Reply:
Changed to layer 3/4 criteria as it is about ip too.